libs/utils/src/postgres_backend_async.rs

```diff
@@ -474,10 +505,35 @@ impl PostgresBackend {
             // We prefer explicit pattern matching to wildcards, because
             // this helps us spot the places where new variants are missing
             FeMessage::CopyData(_) | FeMessage::CopyDone | FeMessage::CopyFail => {
-                bail!("unexpected message type: {:?}", msg);
+                return Err(MaybeIoError::Anyhow(anyhow::anyhow!(
+                    "unexpected message type: {:?}",
+                    msg
+                )));
             }
         }
 
         Ok(ProcessMsgResult::Continue)
     }
 }
+
+pub(super) fn log_query_error(query: &str, e: &MaybeIoError) {
+    match e {
+        MaybeIoError::Io(io) => {
+            if is_expected_io_error(io) {
+                info!("query handler for '{query}' failed with expected io error: {io}");
+            } else {
+                error!("query handler for '{query}' failed with io error: {io}");
+            }
+        }
+        MaybeIoError::Anyhow(e) => {
+            // ":?" uses the alternate formatting style, which makes anyhow display the
+            // full cause of the error, not just the top-level context + its trace.
+            // We don't want to send that in the ErrorResponse though,
+            // because it's not relevant to the compute node logs.
```

Comment:
Is this part of comment still relevant here ?

Reply:
Removed.